consensus/polybft/block_builder_test.go

```diff
@@ -0,0 +1,160 @@
+package polybft
+
+import (
+	"math/big"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/helper/common"
+	"github.com/0xPolygon/polygon-edge/state"
+	itrie "github.com/0xPolygon/polygon-edge/state/immutable-trie"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBlockBuilder_BuildBlockTxOneFailedTxAndOneTakesTooMuchGas(t *testing.T) {
+	const (
+		amount        = 1_000
+		gasPrice      = 1_000
+		gasLimit      = 21000
+		blockGasLimit = 21000 * 3
+		chainID       = 100
+	)
+
+	accounts := [6]*wallet.Account{}
+
+	for i := range accounts {
+		accounts[i] = generateTestAccount(t)
+	}
+
+	forks := &chain.Forks{}
+	logger := hclog.NewNullLogger()
+	signer := crypto.NewSigner(forks.At(0), chainID)
+
+	tmpDir, err := os.MkdirTemp("/tmp", "bbshow")
+	require.NoError(t, err)
+
+	mchain := &chain.Chain{
+		Params: &chain.Params{
+			ChainID: chainID,
+			Forks:   forks,
+		},
+	}
+
+	stateStorage, err := itrie.NewLevelDBStorage(filepath.Join(tmpDir, "trie"), logger)
+	require.NoError(t, err)
+
+	mstate := itrie.NewState(stateStorage)
+	executor := state.NewExecutor(mchain.Params, mstate, logger)
+
+	executor.GetHash = func(header *types.Header) func(i uint64) types.Hash {
+		return func(i uint64) (res types.Hash) {
+			return types.BytesToHash(common.EncodeUint64ToBytes(i))
+		}
+	}
+
+	balanceMap := map[types.Address]*chain.GenesisAccount{}
+
+	for i, acc := range accounts {
+		// the third tx will fail because of insufficient balance
+		if i != 2 {
+			balanceMap[types.Address(acc.Ecdsa.Address())] = &chain.GenesisAccount{
+				Balance: big.NewInt(1_000_000_000_000),
```

Comment:
If value can be arbitrary here, maybe consider using `ethgo.Ether` or `ethgo.Gwei` function here (for the sake of better readability). WDYT?
```suggestion
				Balance: ethgo.Ether(1),
```

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1550/commits/1d61f96bb3d41b1066ea6ddb6dcb7a2804008ab7